Sources/ForageSDK/Foundation/Network/LiveForageService.swift

```diff
@@ -133,6 +134,67 @@ class LiveForageService: ForageService {
         let sessionToken = ForageSDK.shared.sessionToken
         let merchantID = ForageSDK.shared.merchantID
 
+        do {
+            let (vaultResponse, forageRequest) = try await collectPinForPayment(
+                pinCollector: pinCollector,
+                paymentReference: paymentReference,
+                action: .capturePayment
+            )
+
+            _ = try await awaitResult { completion in
+                self.pollingService.execute(
+                    vaultResponse: vaultResponse,
+                    request: forageRequest,
+                    completion: completion
+                )
+            }
+
+            return try await awaitResult { completion in
+                self.getPayment(
+                    sessionToken: sessionToken,
+                    merchantID: merchantID,
+                    paymentRef: paymentReference,
+                    completion: completion
+                )
+            }
+        } catch {
+            throw error
+        }
+    }
+
+    func getPayment(sessionToken: String, merchantID: String, paymentRef: String, completion: @escaping (Result<PaymentModel, Error>) -> Void) {
+        do { try provider.execute(model: PaymentModel.self, endpoint: ForageAPI.getPayment(sessionToken: sessionToken, merchantID: merchantID, paymentRef: paymentRef), completion: completion) } catch { completion(.failure(error)) }
+    }
+
+    // MARK: Collect PIN
+
+    func collectPinForDeferredCapture(
+        pinCollector: VaultCollector,
+        paymentReference: String
+    ) async throws -> VaultResponse {
+        do {
+            let collectPinResult = try await collectPinForPayment(
+                pinCollector: pinCollector,
+                paymentReference: paymentReference,
+                action: .collectPin
+            )
+            return collectPinResult.vaultResponse
+        } catch {
+            throw error
+        }
+    }
+
+    /// Common Payment-related prologue across capturePayment and collectPin.
+    /// Both `collectPin` and `capturePayment` involve the same
+    /// preliminerary data retrieval and a trip to the Vault (VGS or Basis Theory) Proxy
+    private func collectPinForPayment(
+        pinCollector: VaultCollector,
+        paymentReference: String,
+        action: VaultAction
+    ) async throws -> (vaultResponse: VaultResponse, forageRequest: ForageRequestModel) {
```

Comment:
I want to eventually deprecate our use of `ForageRequestModel`

See ticket here 👉🏻 
https://linear.app/joinforage/issue/FX-650/refactor-foragerequestmodel-empty-strings